### PR TITLE
Add support to show connection group attribute in legacy versions

### PR DESCRIPTION
**Pull request by NipuniBhagya**
Legacy Versions

1. With custom claim mappings

image

2. Without custom claim mappings - OIDC

image

2. Without custom claim mappings - SAML

image

### Related Issues
- https://github.com/wso2/product-is/issues/19428

### Related PRs
- Related PR `#1` or (None)

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/connections/components/edit/connection-edit.tsx [hunk 1]
```diff
@@ -451,8 +452,8 @@ export const EditConnection: FunctionComponent<EditConnectionPropsInterface> = (
         const isAttributesEnabledForOIDC: boolean = isOidc && UIConfig?.isCustomClaimMappingEnabled;
 
         // Evaluate whether to Show/Hide `Attributes`.
-        if ((attributesForSamlEnabled || shouldShowTab(type, ConnectionTabTypes.USER_ATTRIBUTES))
-            && (IdentityProviderManagementConstants.IDP_TEMPLATE_IDS.OIDC === type && isAttributesEnabledForOIDC)
+        if (((attributesForSamlEnabled || shouldShowTab(type, ConnectionTabTypes.USER_ATTRIBUTES))
+            || (IdentityProviderManagementConstants.IDP_TEMPLATE_IDS.OIDC === type && isAttributesEnabledForOIDC))
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/connection-edit.tsx:456]: Shall we move this `type === template_ids.OIDC` check to `isAttributesEnabledForOIDC` variable for better readability?

```js
const isAttributesEnabledForOIDC: boolean = isOidc 
    && type === IdentityProviderManagementConstants.IDP_TEMPLATE_IDS.OIDC
    && UIConfig?.isCustomClaimMappingEnabled;

```

File: apps/console/src/features/connections/components/edit/settings/attribute-management/uri-attributes-settings.tsx [hunk 2]
```diff
@@ -64,18 +76,51 @@ export const UriAttributesSettings: FunctionComponent<AdvanceAttributeSettingsPr
...
+    useEffect(() => {
+        if (!selectedClaimMappings || selectedClaimMappings.length === 0) {
+            return;
+        }
+
+        setGroupAttribute(getGroupAttribute());
+    }, [ selectedClaimMappings ]);
+
+    useEffect(() => {
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/settings/attribute-management/uri-attributes-settings.tsx:102]:
```suggestion
```

**Review by pavinduLakshan — APPROVED** [review]

File: apps/console/src/features/connections/components/edit/settings/attribute-management/uri-attributes-settings.tsx [hunk 3]
```diff
@@ -64,18 +76,50 @@ export const UriAttributesSettings: FunctionComponent<AdvanceAttributeSettingsPr
...
+        if (!selectedClaimMappings || selectedClaimMappings.length === 0) {
+            return;
+        }
+
+        setGroupAttribute(getGroupAttribute());
+    }, [ selectedClaimMappings ]);
+
+    useEffect(() => {
+        // If the initial role uri is not available, then use the group attribute.
+        updateRole(initialRoleUri ? initialRoleUri : groupAttribute);
```
Comment (JayaShakthi97) [inline apps/console/src/features/connections/components/edit/settings/attribute-management/uri-attributes-settings.tsx:103]: We can use null check here.
```suggestion
        updateRole(initialRoleUri ?? groupAttribute);
```

File: apps/console/src/features/connections/components/edit/settings/attribute-management/uri-attributes-settings.tsx [hunk 4]
```diff
@@ -64,18 +76,50 @@ export const UriAttributesSettings: FunctionComponent<AdvanceAttributeSettingsPr
...
         isSaml,
         [ "data-testid" ]: testId
     } = props;
 
     const { t } = useTranslation();
+    const { UIConfig } = useUIConfig();
+
+    const [ groupAttribute, setGroupAttribute ] = useState<string>("");
+
+    useEffect(() => {
```
Comment (JayaShakthi97) [inline apps/console/src/features/connections/components/edit/settings/attribute-management/uri-attributes-settings.tsx:93]: Shall we add doc comments for added useEffect explaining the purpose?

**Review by JayaShakthi97 — APPROVED** [review]

**Pull request merged**